src/index.ts

```diff
@@ -54,16 +44,14 @@ export class IDX {
   _docLoader: DataLoader<string, Doctype>
   _indexProxy: DoctypeProxy
   _resolver: Resolver
-  _schemas: SchemasAliases
 
-  constructor({ autopin, ceramic, definitions = {}, resolver = {}, schemas }: IDXOptions) {
+  constructor({ autopin, ceramic, definitions = {}, resolver = {} }: IDXOptions) {
     this._autopin = autopin !== false
     this._ceramic = ceramic
     this._definitions = definitions
-    this._schemas = schemas
 
-    this._docLoader = new DataLoader(async (docIds: ReadonlyArray<string>) => {
-      return await Promise.all(docIds.map(async docId => await this._ceramic.loadDocument(docId)))
+    this._docLoader = new DataLoader(async (ids: ReadonlyArray<string>) => {
```

Comment:
It batches the requests made in a single execution frame, it's useful notably because we need to load the IDX document in order to check the key/value.

For example doing `await Promise.all([idx.get('foo'), idx.get('bar')])` without dataloader would make 2 requests to the IDX document and then 1 request for each key, but using dataloader it's only 1 request for the IDX document and 1 for each key.
If we move forwards with refactoring Ceramic I think this logic should live in the Ceramic client, but in the meantime it's I think it's an easy performance improvement to use it here.

Reply:
Ok, thanks for clarifying. Yeah I think it makes sense to put something like this into ceramic.